### DISCUSSION TITLE
Issue about using MPI for Multiapp

### DISCUSSION THREAD

**Discussion by SomnusYu** [idaholab/moose#27705] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hi all,

I am doing the [TensorMechanics] (old version) and two-phase flow module (like THM) coupling through Multiapp. And the TensorMechanics is the main app in my case. 

When I use` ./app_name -i case_name_main.i` to run my case, it will cost 1693 seconds. However, if I use` mpiexec -n 5  ./app_name -i case_name_main.i`,  it will cost 5478 seconds. Further, if I use` mpiexec -n 10  ./app_name -i case_name_main.i`, it will cost 18124 seconds.

Why does using more computing core not cause a decrease in computation time, but result in more computation time? Shouldn't parallel computing improve computational efficiency?

Thanks!

**Comment by GiudGiud** [comment]
Hello

The solvers do not scale to small numbers of DoFs. Using too many processes can slow down the code

You can try to identify whether tensormechanics or THM is being slow with higher number of cores. Then you can limit the number of processes of the multiapp using the max_procs_per_app parameter

Guillaume

**Reply by SomnusYu** [reply]
Thanks!
